File: pkg/satellite/projectmembers.go
// Copyright (C) 2018 Storj Labs, Inc.
// See LICENSE for copying information.

package satellite

import (
	"context"
	"time"

	"github.com/skyrings/skyring-common/tools/uuid"
)

// ProjectMembers exposes methods to manage ProjectMembers table in database.
type ProjectMembers interface {
	// GetByMemberID is a method for querying project members from the database by memberID.
	GetByMemberID(ctx context.Context, memberID uuid.UUID) ([]ProjectMember, error)
	// GetByProjectID is a method for querying project members from the database by projectID, offset and limit.
	GetByProjectID(ctx context.Context, projectID uuid.UUID, limit int, offset int64) ([]ProjectMember, error)
	// Insert is a method for inserting project member into the database.
	Insert(ctx context.Context, memberID, projectID uuid.UUID) (*ProjectMember, error)
	// Delete is a method for deleting project member by memberID and projectID from the database.
	Delete(ctx context.Context, memberID, projectID uuid.UUID) error
}

// ProjectMember is a database object that describes ProjectMember entity.
type ProjectMember struct {
	ID uuid.UUID

	// FK on Users table.
	MemberID uuid.UUID
	// FK on Projects table.
	ProjectID uuid.UUID

	CreatedAt time.Time
}

File: web/satellite/tests/unit/mock/api/projectMembers.ts
// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import { ProjectMemberCursor, ProjectMembersApi, ProjectMembersPage } from '@/types/projectMembers';

/**
 * Mock for ProjectMembersApi
 */
export class ProjectMembersApiMock implements ProjectMembersApi {
    public cursor: ProjectMemberCursor;
    public page: ProjectMembersPage;

    public setMockPage(page: ProjectMembersPage): void {
        this.page = page;
    }

    add(_projectId: string, _emails: string[]): Promise<void> {
        throw new Error('not implemented');
    }

    delete(_projectId: string, _emails: string[]): Promise<void> {
        throw new Error('not implemented');
    }

    get(_projectId: string, _cursor: ProjectMemberCursor): Promise<ProjectMembersPage> {
        return Promise.resolve(this.page);
    }
}
